### DISCUSSION TITLE
Notifications

### DISCUSSION THREAD

**Discussion by Swallowtail23** [ep1cman/unifi-protect-backup#62] [General]
The buffer size causing unifi-protect-backup to hang for me on the 342MB video got me thinking... No matter the limit, at some point someone is going to have a Unifi video that is too big...

Feature request:
As this is ideally run "headless", configure options for SMTP alert notification, or some other way for push notifications in event of a failed download/upload, or perhaps a "daily digest" style end of day container/app health update/stats summary to ensure ongoing successful function / recovery from failure.

**Comment by ep1cman** [comment]
Hmm this is an interesting idea. I wonder if there is some way to utilise an external tool to support this because there is a huge number of notification services. Maybe it could hook into dockers built in health check system

**Reply by Swallowtail23** [reply]
This is literally the only container I run, so I'm not going to be much help there with any experience in that regard, but happy to test.

**Comment by ep1cman** [comment]
This looks like a possible way to do it: https://notifiers.readthedocs.io/en/latest/index.html

**Comment by ep1cman** [comment]
https://github.com/caronc/apprise < Even better! This will definitely make its way in when I next sit down to add features

**Comment by Swallowtail23** [comment]
This is why it's a good idea. I have the container set to automatically reboot if it sees "Cannot connect to host" with the IP. The NVR had locked up, it rebooted, but still failed to connect on reboot... that was 3 weeks ago...
![image](https://user-images.githubusercontent.com/8214128/218341371-1667faed-9b91-4691-89eb-6309d49a36b5.png)
I had not noticed this.

**Comment by ep1cman** [comment]
Hey, I have just made a first pass at implementing this can you try out the dev branch and let me know if it works out for you?

**Reply by Swallowtail23** [reply]
Travelling for work at the moment, but will do when I'm back, mid March.

**Comment by ep1cman** [comment]
Closing this topic as this is now implemented, and multiple users have tested it to be working
